docs/src/components/icons-list.astro

```diff
@@ -9,15 +9,33 @@ const icons = Object.keys(Icons) as (keyof typeof Icons)[];
 	{
 		icons.map((icon) => {
 			return (
-				<div class="icon-preview">
+				<button class="icon-preview" aria-label={`Copy ${icon} icon to clipboard`} id={icon}>
 					<Icon name={icon} size="1.5rem" />
-					<span>{icon}</span>
-				</div>
+					<span aria-live="polite">{icon}</span>
+				</button>
 			);
 		})
 	}
 </div>
 
+<script>
+  const icons = document.querySelectorAll('.icon-preview');
+  icons.forEach((icon) => {
+    icon.addEventListener('click', () => {
+      const iconName = icon.id;
+      const copiedValue = `<Icon name="${iconName}" size="2rem" />`;
+      navigator.clipboard.writeText(copiedValue);
+
+      const iconLabel = document.querySelector(`#${iconName} span`);
+      iconLabel!.textContent = 'Copied!';
+
+      setTimeout(() => {
+        iconLabel!.textContent= iconName;
+      }, 1000);
```

Comment:
Might be best to avoid the `!` assertion here and just move the related code behind an `if` check. That way if for some reason the span isn’t found, nothing crashes.

We can also update the `querySelector` usage to query directly inside the `icon` which both simplifies our code and should perform a bit better. I’ve updated it to explicitly look for our `aria-live` attribute — that means our visual behaviour should break if we break our screen reader behaviour, which is a good way to help prevent accessibility regressions.

```suggestion
      const iconLabel = icon.querySelector('[aria-live]');
      if (iconLabel) {
        iconLabel.textContent = 'Copied!';
        setTimeout(() => {
          iconLabel.textContent = iconName;
        }, 1000);
      }
```